### ISSUE TITLE
Perfherder test signature keeps changing

### ISSUE THREAD

**Issue by shinglyu** [shinglyu/servo-perf#15]
The page load time test result does not show up on the same graph. Based on @wlach 's comment in https://github.com/servo/servo/issues/10452 : 

> No the performance signature is distinct from the job_guid. The signature is calculated based on the properties of PERFHERDER_DATA (suite name, test name, options) as well as various reference data from the job (machine platform, options, ...). And yes, if you change the performance framework you'll get a new series (though the signature should remain the same, since performance framework is not currently incorporated into the signature).

>  Code for signature calculation is here (warning: it has some rough edges): https://github.com/mozilla/treeherder/blob/master/treeherder/etl/perf.py

**Comment by shinglyu** [comment]
@wlach: After some investigation, I believe the problem is because my subtests list is changing. If a test breaks or timeout, I simply omit them in the performance artifacts. So perhaps I should keep the list of subtests constant, and use some placeholder value like `-1` if the subtest fails? And if I wants to add new subtests in the future, it will break again, right? So I should probably add all the potential subtests now.

**Comment by wlach** [comment]
Yes, that's probably it. But how can you rely on the summary value in that case, if a subtest fails? I think you should just turn your job red/orange (or whatever the servo equivalent is) in that case. Either that or just don't generate summary values.

It's not that anything "breaks" if you add/remove a subtest, Perfherder just generates a new series (though I am reconsidering that behaviour, see https://bugzilla.mozilla.org/show_bug.cgi?id=1266183).

**Comment by shinglyu** [comment]
Thanks! I'll turn the job orange. But I'll keep calculating the summary based on the passed tests. https://github.com/servo/servo/issues/11087 makes about half of the test timeout, so we might have to wait a long time before we have any data available for tracking perf regressions.

**Comment by wlach** [comment]
Actually, if the tests are timing out, I would just disable them. How are you calculating the summary value?

I can try to prioritize this bug if it would be helpful: https://bugzilla.mozilla.org/show_bug.cgi?id=1266183
A lot of people have complained about it...

**Comment by shinglyu** [comment]
@wlach Due to some infinite reflow bug in Servo, I disabled about half of the test cases. But like 50% of the time an enabled test will timeout. If I don't calculate the summary based on the remaining passing tests, we'll have too few data to look at. I know that it don't make much scientific sense to use that data, but I guess my priority is to have at least some data and raise the awareness of the performance issue.

I'd appreciate you working on bug 1266183. I can do some workaround for now, so take your time.
